include/maidsafe/crux/socket.hpp

```diff
@@ -253,34 +254,32 @@ inline void socket::close() {
     keepalive_timer.stop();
     transmit_queue.shutdown();
 
+    auto m = multiplexer;
```

Comment:
Minor suggestion: this technique ia also known as pinning, so you may consider renaming `m` to `pin` to hint at the technique.